assets/javascripts/discourse/components/tc-message-collapser.js

```diff
@@ -0,0 +1,34 @@
+import Component from "@ember/component";
+import { action } from "@ember/object";
+import escape from "discourse-common/lib/escape";
+
+export default Component.extend({
+  collapsed: false,
+  cooked: null,
+  link: null,
+  title: null,
+
+  init() {
+    this._super(...arguments);
+
+    const cookedElement = new DOMParser().parseFromString(
+      this.cooked,
+      "text/xml"
+    ).firstChild;
+    const title = cookedElement.getAttribute("data-youtube-title");
```

Comment:
```suggestion
    const title = cookedElement.dataset.youtubeTitle;
```

Reply:
Unfortunately this isn't possible as `firstChild` is a `Node` not a `HTMLElement`. Even `firstElementChild` doesn't return an element but a document T_T

You can try it out

```
const s = '<div data-youtube-id="WaT_rLGuUr8" data-youtube-title="Japanese Katsu Curry (Pork Cutlet)"></div>';
const t = new DOMParser().parseFromString(s, "text/xml");
t.firstChild.getAttribute("data-youtube-title") // 'Japanese Katsu Curry..
t.firstChild.dataset.youtubeTitle // error
t.firstElementChild.dataset.youtubeTitle // error
```

Reply:
and what about using discourse helper `domFromString`: https://github.com/discourse/discourse/blob/main/app/assets/javascripts/discourse-common/addon/lib/dom-from-string.js


I'm actually not sure what would be the difference between the two solutions here 🤔